tests/rptest/tests/cloud_storage_scrubber_test.py

```diff
@@ -231,22 +329,26 @@ def ntpr_missing_spill_manifest():
             if ntpr not in anomalies_per_ntpr:
                 return False
 
-            if anomalies_per_ntpr[ntpr] is None:
-                return False
+            found = to_delete.path in anomalies_per_ntpr[
+                ntpr].missing_spillover_manifests
+            return found, to_delete.path
 
-            if "missing_spillover_manifests" not in anomalies_per_ntpr[ntpr]:
-                return False
+        missing_spill = wait_until_result(
+            ntpr_missing_spill_manifest,
+            timeout_sec=self.scrub_timeout,
+            backoff_sec=2,
+            err_msg="Missing spillover manifest anomaly not reported")
 
-            return to_delete.path in anomalies_per_ntpr[ntpr][
-                "missing_spillover_manifests"]
+        if ntpr not in expected_anomalies:
```

Comment:
I see this repeated at least two times. Wondering if you considered something like

```py
expected_anomalies = collections.defaultdict(Anomalies.make_empty)
```

Not expecting any changes here. Sharing in case you didn't know about it.

Reply:
Ah yeh. That's neat. I'll touch this test again in the next pr. Will update there.